data_model/models.py

```diff
@@ -0,0 +1,109 @@
+from django.contrib.postgres import fields as pg_fields
+from django.db import models
+
+
+class IETFReport(models.Model):
+    rrname = models.CharField(max_length=100)
+    rrtype = models.CharField(max_length=100)
+    rdata = pg_fields.ArrayField(models.CharField(max_length=100))
+    time_first = models.DateTimeField()
+    time_last = models.DateTimeField()
+
+
+class BaseDataModel(models.Model):
+    evaluation = models.CharField(
+        max_length=100, null=True
+    )  # classification/verdict/found/score/malscore
+    # HybridAnalysisObservable (verdict), BasicMaliciousDetector (malicious),
+    # GoogleSafeBrowsing (malicious), Crowdsec (classifications),
+    # GreyNoise (classification), Cymru (found), Cuckoo (malscore),
+    # Intezer (verdict/sub_verdict), Triage (analysis.score),
+    # HybridAnalysisFileAnalyzer (classification_tags)
+    external_references = pg_fields.ArrayField(
+        models.URLField(), null=True
+    )  # link/external_references/permalink/domains
+    # Crowdsec (link), UrlHaus (external_references), BoxJs,
+    # Cuckoo (result_url/permalink), Intezer (link/analysis_url),
+    # MalwareBazaarFileAnalyzer (permalink/file_information.value), MwDB (permalink),
+    # StringsInfo (data), Triage (permalink), UnpacMe (permalink), XlmMacroDeobfuscator,
+    # Yara (report.list_el.url/rule_url), Yaraify (link),
+    # HybridAnalysisFileAnalyzer (domains),
+    # VirusTotalV3FileAnalyzer (data.relationships.contacted_urls/contacted_domains)
+    related_threats = pg_fields.ArrayField(
+        models.CharField(max_length=100), null=True
+    )  # threats/related_threats
+    # GoogleSafeBrowsing, QuarkEngineAPK (crimes.crime)
+    malware_family = models.CharField(
+        max_length=100, null=True
+    )  # family/family_name/malware_family
+    # HybridAnalysisObservable, Intezer (family_name), Cuckoo, MwDB,
+    # Triage (analysis.family), UnpacMe (results.malware_id.malware_family),
+    # VirusTotalV3FileAnalyzer
+    # (attributes.last_analysis_results.list_el.results/attributes.names)
+    additional_info = (
+        models.JSONField()
+    )  # field for additional information related to a specific analyzer
+
+
+class DomainDataModel(BaseDataModel):
+    ietf_report = models.ForeignKey(
+        IETFReport, on_delete=models.CASCADE, null=True
+    )  # pdns
+    rank = models.IntegerField(null=True)  # Tranco
```

Comment:
I would suggest something like `score`, that is maybe a little more generic and could be used by analyzers that provide a numeric evaluation

Reply:
I would not change the "rank" name cause it is very specific to ranking engines and is the right way to call it.
An additional "score" can be added if there are specific use cases.

Reply:
If the field is named `score` I would suggest to move it in the parent class so all analyzers can set it